### PR TITLE
Sample Kind

### PR DESCRIPTION
**Pull request by jleblond**
Goes with https://github.com/c3g/freezeman_server/pull/144

 do not merge right after approval

### CODE REVIEW

File: src/components/filters/descriptions.js [hunk 1]
```diff
@@ -1,13 +1,12 @@
-import {FILTER_TYPE, BIOSPECIMEN_TYPE, SEX, TAXON} from "../../constants";
+import {FILTER_TYPE, SAMPLE_KIND, SEX, TAXON} from "../../constants";
```
Comment (romgrk) [inline src/components/filters/descriptions.js:1]: Not used

File: src/components/samples/SamplesListContent.js [hunk 2]
```diff
@@ -25,13 +25,15 @@ import FiltersWarning from "../filters/FiltersWarning";
...
-      title: "Type",
-      dataIndex: "biospecimen_type",
+      title: "Sample Kind",
+      dataIndex: "sample_kind__name",
       sorter: true,
       width: 80,
-      render: (type) => <Tag>{type}</Tag>,
+      options: sampleKinds.map(x => ({ label: x.name, value: x.name })), // for getFilterProps
+      render: (_, sample) =>
+        <Tag>{sampleKinds.find(sk => sk.id === sample.sample_kind).name}</Tag>,
```
Comment (romgrk) [inline src/components/samples/SamplesListContent.js:36]: Pass `sampleKindsByID` instead of finding by ID through an array.

Comment (romgrk) [reply src/components/samples/SamplesListContent.js:36]: If you want you can pass `sampleKinds: state.sampleKinds` in `mapStateToProps` and use `sampleKinds.items` and `sampleKinds.itemsByID` here.

File: src/utils/api.js [hunk 3]
```diff
@@ -59,6 +59,10 @@ const api = {
     search: q => get("/samples/search/", { q }),
   },
 
+  sampleKinds: {
+    list: () => get("/sample-kinds/"),
```
Comment (romgrk) [inline src/utils/api.js:63]: No pagination to handle here? If there is pagination, is the number of sample kinds low & expected to stay the same?

**Review by romgrk — APPROVED** [review]

File: src/utils/api.js (hunk 3 above)
Comment (jleblond) [reply src/utils/api.js:63]: In a similar way to ContainerKind, SampleKind is not expected to get much larger. Also, it is not something that will be manageable by interface users. If it happens, I can still add pagination.

**Pull request merged**